src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp

```diff
@@ -1027,49 +1027,43 @@ namespace
         const TimeValue         time)
     {
         // Create environment EDF.
-        std::string env_tex_instance_name;
         if (rend_params.envMap->IsSubClassOf(AppleseedEnvMap::get_class_id()))
         {
             auto appleseed_envmap = static_cast<AppleseedEnvMap*>(rend_params.envMap);
+            appleseed_envmap->Update(time, FOREVER);
+            auto env_map = appleseed_envmap->create_envmap("environment_edf");
+
             IParamBlock2* param_block = appleseed_envmap->GetParamBlock(0);
 
             INode* sun_node(nullptr);
             BOOL sun_node_on(FALSE);
             get_paramblock_value_by_name(param_block, L"sun_node", time, sun_node, FOREVER);
             get_paramblock_value_by_name(param_block, L"sun_node_on", time, sun_node_on, FOREVER);
 
-            float sun_theta, sun_phi;
             if (sun_node != nullptr && sun_node_on)
             {
                 Matrix3 sun_transform = sun_node->GetObjTMAfterWSM(time);
                 sun_transform.NoTrans();
 
                 const Point3 sun_dir = (Point3::ZAxis * sun_transform).Normalize();
-                sun_theta = std::acosf(sun_dir.z);
+                float sun_theta = std::acosf(sun_dir.z);
 
                 float cos_sun_phi = sun_dir.x / sqrt(1.0f - sun_dir.z * sun_dir.z);
                 cos_sun_phi = asf::clamp(cos_sun_phi, -1.0f, 1.0f);
-                sun_phi = std::acosf(cos_sun_phi);
+                float sun_phi = std::acosf(cos_sun_phi);
 
                 if (sun_dir.y > 0.0f)
                     sun_phi = asf::TwoPi<float>() - sun_phi;
 
-                sun_theta = asf::rad_to_deg(sun_theta);
-                sun_phi = asf::rad_to_deg(sun_phi);
-            }
-            else
-            {
-                get_paramblock_value_by_name(param_block, L"sun_theta", time, sun_theta, FOREVER);
```

Comment:
So this code is no longer needed?

Reply:
Yes, `Update()` does that now.
Here's what docs say about material's `Update()` function:
>....This method is called before rendering begins to allow the plug-in to evaluate anything prior to the render so it can store this information. .... It is generally called once at the start of each frame, or during interactive playback when the time changes.
So calling update before the rendering seems correct.